pandas/core/groupby/generic.py

```diff
@@ -413,7 +413,25 @@ def _wrap_transformed_output(
         assert isinstance(result, Series)
         return result
 
-    def _wrap_applied_output(self, keys, values, not_indexed_same=False):
+    def _wrap_applied_output(
+        self, keys: Index, values: List[Any], not_indexed_same: bool = False
```

Comment:
Are values actually `List[Any]` here? Thought we passed ndarrays as well

Reply:
@WillAyd I added
```python
assert isinstance(values, list) or values is None
```
and ran
```bash
$ pytest pandas/tests/groupby/ --pdb
```
, which passed - is there anywhere else that would pass an `ndarray`?

(this comment got me to correct it from `List` to `Optional[List]`, so thanks for that!)

Reply:
Sounds like I was wrong on that then :-) thanks for checking